### PR TITLE
SRCH-3615 Add facet field filtering to Results API

### PR DESCRIPTION
**Pull request by lorawoodford**
## Summary
- Extends audience, content_type, mime_type, and searchgov_custom facet filtering filtering to the Results API (tag filtering is covered in SRCH-3614);
- Many of the changes here replicate rubocop/formatting changes introduced in SRCH-3614 to aid in avoiding merge conflicts regardless of which PR goes into main first;
- I've tried to highlight actual code changes in files loaded up with rubocopy/codeclimate-inspired tweaks;
- Instructions on using the Results API at: https://open.gsa.gov/api/searchgov-results/
 
### Checklist
Please ensure you have addressed all concerns below before marking a PR "ready for review" or before requesting a re-review. If you cannot complete an item below, replace the checkbox with the ⚠️ `:warning:` emoji and explain why the step was not completed.
 
#### Functionality Checks
 
- [x] You have merged the latest changes from the target branch (usually `main`) into your branch.
  
- [x] Your primary commit message is of the format **SRCH-#### \<description\>** matching the associated Jira ticket.

- [x] PR title is either of the format **SRCH-#### \<description\>** matching the associated Jira ticket (i.e. "SRCH-123 implement feature X"), or **Release - SRCH-####, SRCH-####, SRCH-####** matching the Jira ticket numbers in the release.
 
- :warning: Automated checks pass. If Code Climate checks do not pass, explain reason for failures: 
One Code Climate structure engine error remains in `app/models/i14y_search.rb` for "`facet_filter_options` has a Cognitive Complexity of 6".  The cognitive complexity is high due to all the `if` conditionals.  I propose we ignore/approve this error as any adjustments to this method will arguably make it less readable.  I've also temporarily disabled one cop (AbcSize) inline in `app/models/i14y_search.rb` and `lib/api/non_commercial_search_options.rb` as work continues on this functionality and I want to avoid continuing to request Code Climate approvals.
 
#### Process Checks

- [x] You have specified at least one "Reviewer".

### CODE REVIEW

File: spec/controllers/api/v2/searches_controller_spec.rb [hunk 1]
```diff
@@ -276,45 +274,54 @@
...
-          limit: 20,
-          next_offset_within_limit: true,
-          offset: 0,
-          query: 'api',
-          query_not: 'excluded',
-          query_or: 'alternative',
-          query_quote: 'barack obama',
-          sort_by: 'date'
-        })
+        expect(assigns(:search_options).attributes).to include({ access_key: 'basic_key',
```
Comment (lorawoodford) [inline spec/controllers/api/v2/searches_controller_spec.rb:300]: Only substantive code change in spec file.  Remaining copy rubocop/code climate changes added in SRCH-3614.

File: spec/models/i14y_search_spec.rb [hunk 2]
```diff
@@ -1,11 +1,30 @@
+# frozen_string_literal: true
```
Comment (lorawoodford) [inline spec/models/i14y_search_spec.rb:1]: All changes/additions replicate those added in SRCH-3614.  No net-new code here.

File: spec/models/news_search_spec.rb [hunk 3]
```diff
@@ -1,8 +1,8 @@
+# frozen_string_literal: true
```
Comment (lorawoodford) [inline spec/models/news_search_spec.rb:1]: All changes/additions replicate those added in SRCH-3614. No net-new code here.

**Comment by lorawoodford** [comment]
@danswick The upstream changes from SRCH-3614/main have been merged in and this is ready for your review whenever.

File: spec/controllers/api/v2/searches_controller_spec.rb [hunk 4]
```diff
@@ -305,6 +305,42 @@
                                                                  sort_by: 'date' })
       end
 
+      context 'when an audience filter is present' do
+        let(:params_with_audience) { search_params.merge(audience: 'everyone') }
+
+        it 'passes the audience filter to its ApiI4ySearch object' do
```
Comment (danswick) [inline spec/controllers/api/v2/searches_controller_spec.rb:311]: Tiny thing, but the new descriptions all say "i4y". I mix up the `1` and the `i` all the time. 

```suggestion
        it 'passes the audience filter to its ApiI14ySearch object' do
```

**Review by danswick — COMMENTED** [review]
These changes all look great to me. And thank you for including inline comments about the CC/rubocop changes! 

I left a very small comment about a spec description typo, but this seems ready to go otherwise.

File: spec/controllers/api/v2/searches_controller_spec.rb (hunk 4 above)
Comment (lorawoodford) [reply spec/controllers/api/v2/searches_controller_spec.rb]: Oh, fun, this was an existing typo that I copy/pasted 5 additional times.  It's fixed all places now.

**Comment by lorawoodford** [comment]
@danswick Typo fixed.  This will need a CC approval as well.  Thanks.

**Review by danswick — APPROVED** [review]
Beautiful! CC issue ✅ approved as well.

**Pull request merged**